sled-agent/src/services.rs

```diff
@@ -211,9 +199,10 @@ pub struct Config {
     /// An optional internet gateway address for external services.
     pub gateway_address: Option<Ipv4Addr>,
 
-    /// The path for the ServiceManager to store information about
-    /// all running services.
-    pub all_svcs_config_path: PathBuf,
+    // The path for the ServiceManager to store information about
+    // all running services.
+    all_svcs_ledger_path: PathBuf,
```

Comment:
Why the change from `config` to `ledger` here? 

Also, if we are going to change the names, we may also want to change the constants to refer to `LEDGER` instead of `CONFIG`.

Reply:
I think I got confused between the following overloaded use of the term "config":

- We use "config" to describe parameters for a variety of modules within the sled agent, to describe tweaks on how they should be executed. For example, the sled agent has a "config", the bootstrap agent has a "config", the storage manager also has a "config".
- **totally separately**, we're storing the list of services that the sled manages. I had previously called this "config", but I think the name was not-totally-accurate, so I'm updating it to ledger. After all, it's just a ledger of "what am I responsible for running".

RE: the constants, will do!

Reply:
Done in https://github.com/oxidecomputer/omicron/pull/2946/commits/5d599516d4e4504bbae00347dcddbeea8f17146a